### PR TITLE
Refactor vesting for prove

### LINKED ISSUE #26: Fix the proof after vesting contract change
This commit https://github.com/Entropy-Foundation/aptos-core/pull/25/commits/d5327c92e4817b4e7a700d461577e7c6f0cd8d65 in #25 remove distribute and distribute_shareholder method. 

The proof in spec need to fixed properly.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #26

### CODE REVIEW

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.move [hunk 1]
```diff
@@ -390,60 +390,69 @@ module supra_framework::vesting_without_staking {
...
+
+        // Index is 0-based while period is 1-based so we need to subtract 1.
+        let schedule_index = *next_period_to_vest - 1;
+        let vesting_fraction = if (schedule_index < vector::length(schedule)) {
+            *vector::borrow(schedule, schedule_index)
+        } else {
+            // Last vesting schedule fraction will repeat until the grant runs out.
+            *vector::borrow(schedule, vector::length(schedule) - 1)
+        };
+
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/vesting_without_staking.move]: I think perhaps this code can either go in a separate function, that takes `next_period_to_vest` and returns `vesting_fraction`, or it can go in the caller `vest_individual` so that this particular function merely does the transfer as per given `vesting_fraction`.

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.move [hunk 2]
```diff
@@ -390,60 +390,69 @@ module supra_framework::vesting_without_staking {
...
+            VestEvent {
+                admin: vesting_contract.admin,
+                shareholder_address: shareholder_address,
+                vesting_contract_address: contract_address,
+                period_vested: *next_period_to_vest,
+            },
+        );
+        //update last_vested_period for the shareholder
+        vesting_record.last_vested_period = *next_period_to_vest;
+        *next_period_to_vest = *next_period_to_vest + 1;
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/vesting_without_staking.move]: This should definitely be in the caller `vest_individual`, because otherwise the loop is very hard to read and reason whether it would even terminate, since increment is happening in the callee.

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move [hunk 3]
```diff
@@ -85,46 +85,60 @@ spec supra_framework::vesting_without_staking {
...
-    //     ensures (len(old(shareholders_address)) == len(old(vesting_records))) ==> (len(shareholders_address) == len(vesting_records));
-    // }
+    spec vest_individual {
+        pragma verify = true;
+        pragma aborts_if_is_partial = true;
+        include VestingContractActive;
+        let vesting_contract_pre = global<VestingContract>(contract_address);
+        let post vesting_contract_post = global<VestingContract>(contract_address);
+        let vesting_schedule = vesting_contract_pre.vesting_schedule;
+        let last_vested_period = vesting_schedule.last_vested_period;
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move]: We are not making use of `last_vested_period` at the contract level, apart from initialization, I am wondering why is this needed here. `last_vested_period` is being maintained per shareholder now.

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move [hunk 4]
```diff
@@ -85,46 +85,60 @@ spec supra_framework::vesting_without_staking {
...
+        let next_period_to_vest = last_vested_period + 1;
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move]: Again, since `last_vested_period` is maintained on per shareholder basis, this doesn't make sense.

**Review by sjoshisupra — CHANGES_REQUESTED** [review]

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move (hunk 4 above)
Comment (ghost) [reply aptos-move/framework/supra-framework/sources/vesting_without_staking.spec.move]: I think this addition at spec level is just store the value of `last_vested_period` + 1. why this is needed? because this is for write the spec for pre and post state stays the same.

**Review by sjoshisupra — APPROVED** [review]

File: aptos-move/framework/supra-framework/sources/vesting_without_staking.move
Comment (francojaf) [inline aptos-move/framework/supra-framework/sources/vesting_without_staking.move]: Const
Address:"0xb22ccca8eaca77e0343ab6357c0d880833afab74da44f2557a85313fa22d9ab1",
